### PR TITLE
WIP: Custom token support

### LINKED ISSUE #1287: [Enhancement] Integrator-configurable arbitrary token bridging
- Enable integrators to add tokens that they want to support in Connect. 
- In this iteration these tokens should use the 2-tx route.
- There needs to be validation to disallow duplicate tokens (among other things)

### PR DESCRIPTION
**Pull request by artursapek**
Addresses #1287

### CODE REVIEW

File: wormhole-connect/src/config/index.ts [hunk 1]
```diff
@@ -13,13 +13,18 @@ import {
...
+  validateConfigs,
+  validateDefaults,
+} from './utils';
 import { Alignment } from 'components/Header';
 
 const el = document.getElementById('wormhole-connect');
 if (!el)
   throw new Error('must specify an anchor element with id wormhole-connect');
 const configJson = el.getAttribute('config');
+console.log(el, configJson);
```
Comment (barnjamin) [inline wormhole-connect/src/config/index.ts]:
```suggestion
```

File: wormhole-connect/src/config/index.ts [hunk 2]
```diff
@@ -59,34 +64,32 @@ export const WORMHOLE_API =
...
 export const EXPLORER = config.explorer;
 
-export const ATTEST_URL =
-  ENV === 'MAINNET'
-    ? 'https://www.portalbridge.com/#/register'
-    : ENV === 'DEVNET'
-    ? ''
-    : 'https://wormhole-foundation.github.io/example-token-bridge-ui/#/register';
+export const ATTEST_URL = {
+  MAINNET: 'https://www.portalbridge.com/#/register',
```
Comment (sebastianscatularo) [inline wormhole-connect/src/config/index.ts]: should be https://portalbridge.com/advanced-tools/#/register

probably would be good to handle it in a separated ticket

File: wormhole-connect/src/config/utils.ts [hunk 3]
```diff
@@ -43,6 +43,46 @@ export const validateChainResources = () => {
...
+    }
+    if (builtinKeys.includes(customToken.key)) {
+      console.warn(
+        `Skipping custom token config for "${key}" because its key "${customToken.key}" conflicts with a built-in`,
+      );
+      continue;
+    }
+
+    // Accept custom token config
+    console.warn(`Accepted custom token config for "${key}"`);
```
Comment (sebastianscatularo) [inline wormhole-connect/src/config/utils.ts]: seems odd a warning log level for a success process

File: wormhole-connect/src/icons/TokenIcons.tsx [hunk 4]
```diff
@@ -40,125 +40,69 @@ const useStyles = makeStyles<{ size: number }>()((theme, { size }) => ({
...
-      return PYTH;
-    }
-    case Icon.KLAY: {
-      return KLAY;
-    }
-    default: {
-      return noIcon;
-    }
-  }
+const iconMap: { [key in Icon]: React.JSX.Element } = {
```
Comment (sebastianscatularo) [inline wormhole-connect/src/icons/TokenIcons.tsx:53]: ❤️

**Comment by Eiei070922** [comment]
Thank you. You guys don't have to send me the documents. I haven't played and played for over a year, wasting a lot of time and getting nothing, only bad things. Thank you very much. I'll take my money back if I owe you anything🙏

รับ Outlook สำหรับ Android<https://aka.ms/AAb9ysg>
________________________________

File: wormhole-connect/src/config/utils.ts (hunk 3 above)
Comment (artursapek) [reply wormhole-connect/src/config/utils.ts]: good point lol

**Review by sebastianscatularo — APPROVED** [review]

**Comment by crypt0xg** [comment]
👍

**Pull request merged**